login_routes/router.js

```diff
@@ -2,21 +2,152 @@ const express = require('express');
-//need to set up a post request to create a new driver account with the google id token and the user information that is input on the front end
-//need to set up a post request to create a new valet account with the google id token and the user information that is input on the front end (might combine these requests into one for more dry code)
+//these are the routes that i will need to set up.
+//driver/valet account creation, get account info
 
-router.get('/login/test', async (req, res) => {
+//get the account info and return null if there is nothing there
+router.get('/login/:id', async (req, res) => {
   try {
-    const { data, error } = await accountInfo.select('*');
+    const { id } = req.params;
+    const { data, error } = await supabase
+      .from('accounts')
+      .select()
+      .eq('id', id);
+    if (error) {
+      console.error('Error retrieving user data:', error);
+      return res.sendStatus(500);
+    }
+    if (!data) {
+      return res.status(201).send(null);
+    }
+    res.status(200).json(data);
+  } catch (error) {
+    console.error('Error retrieving user data:', error);
+    res.status(500).send('Internal Server Error');
+  }
+});
+
+//create the driver account, and the car associated with that account in the database.
+//will probably refactor to be a supabase transaction for atomicity, so that if one fails, the other will fail as well. so we dont have to deal with a partially constructed record in the database.
+router.post('/login/:id/driver', async (req, res) => {
+  const { id } = req.params;
+  const { google_accounts_id, first_name, last_name, email, phone_number, role, make, model, color, license_plate_number } = req.body;
+  try {
+    const { error: accountsError } = await supabase
+      .from('accounts')
+      .insert(
+        {
+          id,
+          first_name,
+          last_name,
+          email,
+          phone_number,
+          role
+        }
+      );
+    if (accountsError) {
+      console.error('Error creating driver account:', accountsError);
+      return res.sendStatus(500);
+    }
+    const { error: carsError } = await supabase
+      .from('cars')
+      .insert(
+        {
+          make,
+          model,
+          color,
+          license_plate_number,
+          user_id: id
+        }
+      );
+    if (carsError) {
+      console.error('Error creating driver account (cars):', carsError);
+      return res.sendStatus(500);
+    }
+    res.sendStatus(201);
+  }
+  catch (error) {
+    console.error('Error creating account:', error);
+    res.status(500).json({ error: 'Internal server error' });
+  }
+
+});
+//need to set up a post request to create a new valet account with the google id token and the user information that is input on the front end. this also assumes that a middleware function is used that generates the lat and lng from the passed in address. then creates the garage and the parking spots associated with that garage.
```

Comment:
Here seems like there are two different forms, one for drivers and one for valet. Okay with this implementation